tools/wiki_entities/high_quality_phrase_filter.py

```diff
@@ -39,11 +39,14 @@ def Load(filename, stopwords, output_filename):
             continue
         tokens = line.strip().split('\t')
         valid = False
-        for token in tokens[2:]:
+        for token in tokens[3:]: # First 3 are name and metadata
             support = int(token.split(':')[-2])
             percentage = float(token.split(':')[-1][:-1])
             if (percentage >= MIN_PERCENT) or (support >= MIN_SUP):
-                name = ':'.join(token.split(':')[:-2])
+
+                # Q2736:Association football:15616:0:84% => Association football
+                name = ':'.join(token.split(':')[1:-3]) 
```

Comment:
Could you also check in a sample entities file for verification? I am not sure why we made this error but still derived the output wiki_*.txt in the first place.

Reply:
Hey! Probably because of this:
```
def NoSeparator(name):
    for ch in ',;:()':
        if name.find(ch) != -1:
            return False
    return True

...
if NoSeparator(name) and StopWordChecking(name, stopwords):
    candidate.add(name.lower())
```
No names with `:` were added for the `wiki_quality.txt` script, thus this was unnoticeable.

However, when testing with https://github.com/kno10/WikipediaEntities/blob/master/wiki-20151202/entities.gz and `wiki_all.txt`, it is noticeable:
```
evolutionarybiology
q5025812:camden school for girls
live1966
analysegénétique
buntmetallerz
q1049203:otome game
q4826719:automotive service excellence
q1349876:photek
....
```

Reply:
I will write another commit overwriting the previous one ;)

Reply:
Hey I found one entry which doesn't follow the regex rule. I am implementing both of them and the version will be decided iff both (regex and : count) are either true of false.